app/middlewares/users.js

```diff
@@ -0,0 +1,24 @@
+const { body } = require('express-validator');
+
+const { fieldErrorsValidation } = require('./fieldErrors');
+const { findUserByEmail } = require('../services/users');
+
+exports.userBodyValidations = fieldErrorsValidation([
+  body('first_name', 'first_name must be present')
+    .not()
+    .isEmpty(),
+  body('last_name', 'last_name must be present')
+    .not()
+    .isEmpty(),
+  body('password', 'password must be present')
+    .not()
+    .isEmpty(),
+  body('email')
+    .isEmail()
+    .withMessage('email must have email format')
+    .bail()
+    .custom(async email => {
+      const user = await findUserByEmail(email);
+      if (user) throw new Error('E-mail already in use');
+    })
```

Comment:
This validation should be in another function inside this file. For two reasons:
- This are all 'sintactic' validations, so it not fits right here
- It's very common that other endpoints make the same validation (does this user exists?). I imagine that all the actions that will affect users, will check if the user exists.